### PR TITLE
[Phase-2 L1T] small fixes for RCT cluster emulation

### PR DESCRIPTION
**Pull request by pallabidas**
#### PR description:

Small changes to correctly use ECAL LSB = 0.5 in firmware based code. A fix in RCT cluster stitching function and shower shape flag is implemented.

Also being reviewed here: https://github.com/cms-l1t-offline/cmssw/pull/1206

#### PR validation:

Passes local validation checks with efficiency and rate plots, to be presented in a DP note.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

not a backport

@aloeliger @epalencia

### CODE REVIEW

File: L1Trigger/L1CaloTrigger/interface/Phase2L1CaloEGammaUtils.h [hunk 1]
```diff
@@ -928,19 +922,23 @@ namespace p2eg {
     return is_iso;
   }
 
-  inline bool passes_ss(float pt, float ss) {
+  inline bool passes_ss(float pt, float eta, float ss) {
     bool is_ss;
     if (pt > 130)
       is_ss = true;
+    else if ((abs(eta) > 0.4 && abs(eta) < 0.5) || (abs(eta) > 0.75 && abs(eta) < 0.85))  // temporary adjustment
+      is_ss = ((loose_ss_offset + c1_ss * std::exp(-c2_ss * pt)) <= ss);
```
Comment (srimanob) [inline L1Trigger/L1CaloTrigger/interface/Phase2L1CaloEGammaUtils.h:930]: This part contains many magic numbers, including pt cut above. Is it possible to define them as `constexpr float` before?

Comment (pallabidas) [reply L1Trigger/L1CaloTrigger/interface/Phase2L1CaloEGammaUtils.h:932]: @srimanob done

**Comment by epalencia** [comment]
Please test

**Comment by aloeliger** [comment]
+l1
- Already merged in l1 fork as part of phase 2 prototype

**Comment by srimanob** [comment]
+Upgrade

**Comment by pallabidas** [comment]
Hi @srimanob we would like this to be included in the new samples, so yes for backport to 14_0

**Comment by antoniovilela** [comment]
+1

**Pull request merged**